### PR TITLE
Scripts

### PR DESCRIPTION
**Pull request by maweigert**
A

### CODE REVIEW

File: stardist/scripts/predict2d.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+""")
+    parser.add_argument("-i","--input", type=str, nargs="+", required=True, help = "input file (tiff)")
+    parser.add_argument("-o","--outdir", type=str, default='.', help = "output directory")
+    parser.add_argument("--outname", type=str, nargs="+", default='{img}.stardist.tif', help = "output file name (tiff)")
+
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('-m', '--model', type=str, default=None, help = "model folder / pretrained model to use")
+    parser.add_argument("--axes", type=str, default = None, help = "axes to use for the input, e.g. 'XYC'")
+    parser.add_argument("--n_tiles", type=int, nargs=2, default = None, help = "number of tiles to use for prediction")
+    parser.add_argument("--pnorm", type=float, nargs=2, default = [3,99.8], help = "pmin/pmax to use for normalization")
```
Comment (uschmidt83) [inline stardist/scripts/predict2d.py:31]: Different from the "default" percentiles we normally use.

File: stardist/scripts/predict2d.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    parser.add_argument("-v", "--verbose", action='store_true')
+        
+    args = parser.parse_args()
+
+
+    from csbdeep.utils import normalize
+    from csbdeep.models.base_model import get_registered_models
+    from stardist.models import StarDist2D
+    from imageio import imread
+    from tifffile import imsave
```
Comment (uschmidt83) [inline stardist/scripts/predict2d.py:44]: use `tifffile.imwrite` instead?

File: stardist/scripts/predict2d.py (hunk 1 above)
Comment (maweigert) [reply stardist/scripts/predict2d.py:31]: feel free to change!

File: stardist/scripts/predict2d.py (hunk 2 above)
Comment (maweigert) [reply stardist/scripts/predict2d.py:44]: yep!

File: stardist/scripts/predict3d.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+"""
+
+Command line script to perform prediction in 2D
```
Comment (uschmidt83) [inline stardist/scripts/predict3d.py:3]:
```suggestion
Command line script to perform prediction in 3D
```

File: stardist/scripts/predict2d.py (hunk 1 above)
Comment (uschmidt83) [inline stardist/scripts/predict2d.py:31]:
```suggestion
    parser.add_argument("--pnorm", type=float, nargs=2, default = [1,99.8], help = "pmin/pmax to use for normalization")
```

File: stardist/scripts/predict2d.py (hunk 2 above)
Comment (uschmidt83) [reply stardist/scripts/predict2d.py:44]:
```suggestion
    from tifffile import imwrite
```

File: stardist/scripts/predict2d.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        img = normalize(img,*args.pnorm)
+
+        labels, _ = model.predict_instances(img,
+                                n_tiles=args.n_tiles,
+                                prob_thresh=args.prob_thresh,
+                                nms_thresh=args.nms_thresh)
+        out = pathlib.Path(args.outdir)
+        out.mkdir(parents=True,exist_ok=True)
+
+        imsave(out/args.outname.format(img=pathlib.Path(fname).with_suffix('').name), labels, compress=3)
```
Comment (uschmidt83) [inline stardist/scripts/predict2d.py:86]:
```suggestion
        imwrite(out/args.outname.format(img=pathlib.Path(fname).with_suffix('').name), labels, compress=3)
```

File: stardist/scripts/predict3d.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+""")
+    parser.add_argument("-i","--input", type=str, nargs="+", required=True, help = "input file (tiff)")
+    parser.add_argument("-o","--outdir", type=str, default='.', help = "output directory")
+    parser.add_argument("--outname", type=str, nargs="+", default='{img}.stardist.tif', help = "output file name (tiff)")
+
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('-m', '--model', type=str, default=None, help = "model folder / pretrained model to use")
+    parser.add_argument("--axes", type=str, default = None, help = "axes to use for the input, e.g. 'XYC'")
+    parser.add_argument("--n_tiles", type=int, nargs=3, default = None, help = "number of tiles to use for prediction")
+    parser.add_argument("--pnorm", type=float, nargs=2, default = [3,99.8], help = "pmin/pmax to use for normalization")
```
Comment (uschmidt83) [inline stardist/scripts/predict3d.py:31]:
```suggestion
    parser.add_argument("--pnorm", type=float, nargs=2, default = [1,99.8], help = "pmin/pmax to use for normalization")
```

**Review by uschmidt83 — APPROVED** [review]

**Pull request merged**